Sub: implement set_target_angle_and_climbrate for scripting

Williangalvani opened the pull request.

ES-Alexander reviewed (approved):
The code makes sense to me (but I haven't tested the functionality).

Could be good to add an autotest to make sure it actually works as expected.

I don't love that we convert from euler to quaternion and then internally convert back again, but it makes sense that users will probably prefer to specify euler angles, and we're probably just behind on the `set_guided` implementation or something (and that's out of scope for this PR).

Williangalvani committed:
Sub: implement set_target_angle_and_climbrate for scripting

Williangalvani committed:
Sub: implement set_target_velocity_NED for scripting

allows scripting to command forward/lateral/vertical velocity in
guided mode by feeding the guided angle controller's open-loop
forward and lateral thrust outputs.

Williangalvani committed:
autotest: test Sub guided-mode scripting bindings

Add SITL tests for the ArduSub set_target_angle_and_climbrate and
set_target_velocity_NED scripting bindings. Each binding is driven by a
small Lua helper script that exposes its commands through a dedicated
parameter table so the test can command attitude, climb rate and
velocity at runtime on the vectored_6dof frame.
